From d437e6ba34405fed929ff7aad544bf6da3395d6e Mon Sep 17 00:00:00 2001
From: Daiki Ueno <ueno@gnu.org>
Date: Thu, 26 Feb 2026 08:19:52 +0900
Subject: [PATCH 06/11] BACKPORT: dsa-fips: omit digest_size argument for
 sha384_digest with Nettle 4

Signed-off-by: Daiki Ueno <ueno@gnu.org>
(cherry picked from commit fa048f64445c61643938450cc78994b4e519b399)
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 lib/nettle/int/dsa-fips.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/nettle/int/dsa-fips.h b/lib/nettle/int/dsa-fips.h
index 5c1c90e3b..e0e80c551 100644
--- a/lib/nettle/int/dsa-fips.h
+++ b/lib/nettle/int/dsa-fips.h
@@ -25,6 +25,7 @@
 #include <nettle/bignum.h> /* includes gmp.h */
 #include <nettle/dsa.h>
 #include <nettle/sha2.h>
+#include <nettle/version.h>
 #include "fips.h"
 
 #define div_ceil(x, y) ((x + (y) - 1) / (y))
@@ -100,7 +101,11 @@ inline static void hash(uint8_t digest[DIGEST_SIZE], unsigned length,
 
 	sha384_init(&ctx);
 	sha384_update(&ctx, length, data);
+#if NETTLE_VERSION_MAJOR >= 4
+	sha384_digest(&ctx, digest);
+#else
 	sha384_digest(&ctx, DIGEST_SIZE, digest);
+#endif
 
 	return;
 }
-- 
2.53.0

